src/main/kotlin/org/rust/lang/refactoring/RsImportOptimizer.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.refactoring
+
+import com.intellij.lang.ImportOptimizer
+import com.intellij.psi.PsiFile
+import org.rust.ide.formatter.processors.asTrivial
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.RsElement
+import org.rust.lang.core.psi.ext.RsMod
+import org.rust.lang.core.psi.ext.childrenOfType
+import org.rust.lang.core.psi.ext.descendantsOfType
+import org.rust.openapiext.runWriteCommandAction
+
+class RsImportOptimizer : ImportOptimizer {
+    override fun supports(file: PsiFile?): Boolean = file is RsFile
+
+    override fun processFile(file: PsiFile?) = Runnable {
+        execute(file as RsFile)
+    }
+
+    fun execute(mod: RsMod) {
+        val uses = mod.childrenOfType<RsUseItem>()
+        if (uses.isEmpty()) {
+            return
+        }
+        replaceOrder(mod, uses)
+        val mods = mod.childrenOfType<RsMod>()
+        mods.forEach { execute(it) }
+    }
+
+    companion object {
+
+        private fun optimizeUseSpeck(psiFactory: RsPsiFactory, useSpeck: RsUseSpeck) {
+            if (removeCurlyBraces(psiFactory, useSpeck)) return
+            val useSpeckList = useSpeck.useGroup?.useSpeckList ?: return
+            if (useSpeckList.size < 2) return
+            useSpeckList.forEach { optimizeUseSpeck(psiFactory, it) }
+            val sortedList = useSpeckList
+                .sortedWith(compareByDescending<RsUseSpeck>{ it.path?.self?.text }.thenBy { it.pathText })
```

Comment:
Let's replace `compareByDescending<RsUseSpeck>{ it.path?.self?.text }` by `compareBy<RsUseSpeck> { it.path?.self == null }`. It seems it's a little bit clearer